commit c2fecc58da0e64d22bc86cacf01cbd7aec2c5324
Author: Dmitry Osipenko <dmitry.osipenko@collabora.com>
Date:   Fri Nov 28 02:24:12 2025 +0300

    DRMBackend: Fix NULL dereference in drm_prepare_liftoff()
    
    Fix gamescope crashing with a NULL dereference in drm_prepare_liftoff().
    
    Fixes: 8f82d5153840 ("backend: Add support for deferred backends")
    Signed-off-by: Dmitry Osipenko <dmitry.osipenko@collabora.com>

diff --git a/src/Backends/DRMBackend.cpp b/src/Backends/DRMBackend.cpp
index 048a64f..7318254 100644
--- a/src/Backends/DRMBackend.cpp
+++ b/src/Backends/DRMBackend.cpp
@@ -2561,7 +2561,7 @@ drm_prepare_liftoff( struct drm_t *drm, const struct FrameInfo_t *frameInfo, boo
 		if ( i < frameInfo->layerCount )
 		{
 			const FrameInfo_t::Layer_t *pLayer = &frameInfo->layers[ i ];
-			gamescope::CDRMFb *pDrmFb = static_cast<gamescope::CDRMFb *>( pLayer->tex ? pLayer->tex->GetBackendFb()->Unwrap() : nullptr );
+			gamescope::CDRMFb *pDrmFb = static_cast<gamescope::CDRMFb *>( (pLayer->tex && pLayer->tex->GetBackendFb()) ? pLayer->tex->GetBackendFb()->Unwrap() : nullptr );
 
 			if ( pDrmFb == nullptr )
 			{
